cli/root.go

```diff
@@ -1144,6 +1179,21 @@ func formatCoderSDKError(err *codersdk.Error, opts *formatOpts) string {
 	return str.String()
 }
 
+// traceError is a helper function that aides developers debugging failed cli
+// commands. When we pretty print errors, we lose the context in which they came.
+// This function adds the context back. Unfortunately there is no easy way to get
+// the prefix to: "error string: %w", so we do a bit of string manipulation.
+//
+//nolint:errorlint
+func traceError(err error) string {
+	if uw, ok := err.(interface{ Unwrap() error }); ok {
+		a, b := err.Error(), uw.Unwrap().Error()
+		c := strings.TrimSuffix(a, b)
+		return c
+	}
+	return err.Error()
+}
+
```

Comment:
This is the biggest thing I added, and it only prints in `verbose` mode.

Essentially when we wrap some errors, we lose the context because the pretty printer doesn't know what to do with them.

An example is:

```
return fmt.Errorf("fetch buildinfo: %w", err)
```

The underlying `err` is an SDK error, so the context `fetch buildinfo:` is lost.
This code aims to put that back in somewhere that is reasonable, so if you run `-v` in your command, that info returns.

----

I am debating just always printing this, because I can't image it gets too unwieldly.

Reply:
I decided to always print this. Without it, it is really hard to debug where some api calls fail.